### ISSUE TITLE
update-daemon

### ISSUE THREAD

**Issue by dylang** [sindresorhus/module-requests#5]
Take the logic of https://github.com/yeoman/update-notifier and just do the update behind the scenes. 

Only meant for packages installed globally.

Options are to do the update when the program runs or auto-update as a background task so your globals are always up-to-date.

**Comment by sindresorhus** [comment]
Would you prefer a system daemon that runs once in a while and does it for all global packages or a module that is included as dependency of CLI apps and only does it for that app. Maybe both?

**Comment by dylang** [comment]
Great question @sindresorhus. Maybe both, but included as a dependency seems to me more likely it would be used, since it would just be automatic. 

What do you think?

**Comment by sindresorhus** [comment]
Yeah, I like it. I would probably use the system daemon thing, but I imagine having it as a dependency could be useful for many CLIs. 

Though I must note that it's going to get a lot of resistance. I actually implemented auto-update early on with Yeoman, but had to back down and make update-notifier instead as so many people were against auto-update.

**Comment by icyflame** [comment]
@sindresorhus Would it be something like `apm update`?

`apm update` is awesome, as it lets you see the updates available for all installed packages, and if required, update just some of them. It is especially helpful when there are a lot of packages to update.

**Comment by forivall** [comment]
As long as it's not as slow as apm update. Maybe, like apm update, but daemonize for the actual update process, and then a separate subcommand or another `update` should then show the update status.

**Comment by dylang** [comment]
Maybe showing a list of changes would make people more calm about auto-updating. 

I have a package called [changelog](https://github.com/dylang/changelog) that guestimates changes based on git commits, but I haven't given it much attention in the last year or so:

![screen shot 2015-09-23 at 10 40 18 am](https://cloud.githubusercontent.com/assets/51505/10048134/870431ac-61df-11e5-8b46-40fee63c8c82.png)

**Comment by dylang** [comment]
Another option is to assume `everything always auto-updates`, but let users change this setting on a per-package basis with something like:

```
$ update ask trash
$ update auto trash
$ update never trash
```

And override all with something like

```
$ update ask *
$ update auto *
$ update never *
```

List the config
```
$ update ls
├── trash auto 2.0.0
├── npm never 2.5.15 -> 3.3.3
└── gulp ask 1.3.5 -> 1.3.6
```

[configstore](https://github.com/yeoman/configstore) is a great package for storing settings like this.

*Usability note: This adds more user-facing configuration, which can lead to more confusion and problems.*
